balancer/rls/picker.go

```diff
@@ -252,6 +252,26 @@ func (p *rlsPicker) handleRouteLookupResponse(cacheKey cacheKey, targets []strin
 	// entry would be used until expiration, and a new picker would be sent upon
 	// backoff expiry.
 	now := time.Now()
+
+	// "An RLS request is considered to have failed if it returns a non-OK
+	// status or the RLS response's targets list does not contain any non-empty
+	// entries." - RLS LB Policy design.
+	var nonEmptyTarget bool
+	for _, target := range targets {
+		if target != "" {
+			nonEmptyTarget = true
+			break
+		}
+	}
+	if !nonEmptyTarget {
+		if err == nil {
+			err = fmt.Errorf("RLS response's target list does not contain any non-empty entries for key %+v", cacheKey)
+		} else {
+			// Can eat the status part of error since the RLS doesn't forward
+			// statuses from control plane RPCss to data plane RPCss (see Pick).
+			err = fmt.Errorf(err.Error()+"and also RLS response's target list does not contain any non-empty entries for key %+v", cacheKey)
+		}
```

Comment:
If err is set, don't we expect `targets` to always be empty?  I believe they should be mutually exclusive:

https://github.com/zasweq/grpc-go/blob/e853dbf004c343da4b8c6204524765ba6fbeef38/balancer/rls/control_channel.go#L216

Reply:
Pinged you offline about this. Done.